### PR TITLE
General improvements to SLURM adaptor

### PR DESCRIPTION
**Pull request by marksantcroos**
Specifically to support Comet, also tested on Stampede.

### CODE REVIEW

**Comment by andre-merzky** [comment]
Looks good to me, will try to test/merge today.  Does that require saga/rp releases to be in sync?  I assume so?

**Comment by marksantcroos** [comment]
On stampede it should work without the SAGA changes I think. But for Comet it will need the SAGA changes indeed.

**Comment by andre-merzky** [comment]
Ok, will test both.  Thanks.

File: src/saga/adaptors/slurm/slurm_job.py [hunk 1]
```diff
@@ -516,8 +512,13 @@ def _job_run (self, jd) :
                               % (total_cpu_count, number_of_processes), 
                                  saga.NoSuccess, self._logger)
 
-        slurm_script += "#SBATCH --ntasks=%s\n"        % (number_of_processes)
-        slurm_script += "#SBATCH --cpus-per-task=%s\n" % (total_cpu_count/number_of_processes)
+        slurm_script += "#SBATCH --ntasks=%s\n" % (number_of_processes)
+
+        if total_cpu_count != number_of_processes:
+            slurm_script += "#SBATCH --cpus-per-task=%s\n" % (total_cpu_count / number_of_processes)
```
Comment (andre-merzky) [inline src/saga/adaptors/slurm/slurm_job.py]: Can this be a float?

Comment (marksantcroos) [inline src/saga/adaptors/slurm/slurm_job.py]: No, integer division only.

Comment (andre-merzky) [inline src/saga/adaptors/slurm/slurm_job.py]: ack

**Comment by andre-merzky** [comment]
Thanks Mr. Rabbit!

**Pull request merged**